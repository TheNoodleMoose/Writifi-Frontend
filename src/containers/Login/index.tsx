import React from "react";
import styled from "styled-components";
import LoginCard from "../../components/LoginCard";
import { ReactComponent as ComputerWork } from "../../assets/svgs/computerWork.svg";

const Login = () => {
  return (
    <LoginContainer>
      <WelcomeContainer>
        <WelcomeTitle>Welcome back!</WelcomeTitle>
        <WelcomeDescription>
          Get logged in and start discovering...
        </WelcomeDescription>
        <ComputerWorkSvg width={800} height={600} />
      </WelcomeContainer>
      <LoginCard />
    </LoginContainer>
  );
};

export default Login;

const LoginContainer = styled.div`
  display: flex;
  background-color: aquamarine;
  justify-content: space-around;
  align-items: center;
  height: 95%;
  @media (max-width: 768px) {
    flex-direction: column;
    align-items: center;
    justify-content: space-evenly;
  }
`;

const ComputerWorkSvg = styled(ComputerWork)`
  width: 100%;
  @media (max-width: 768px) {
    display: none;
  }
`;

const WelcomeContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  @media (max-width: 768px) {
    width: 70%;
  }
`;

const WelcomeTitle = styled.h3`
  font-size: 4em;
  font-weight: bold;
  color: #2c3e50;
`;

const WelcomeDescription = styled.p`
  font-size: 2em;
  color: #2c3e50;
  font-weight: 500;
`;
