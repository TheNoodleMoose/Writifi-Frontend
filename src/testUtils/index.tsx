import React, { ReactElement, ReactNode } from "react";

import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export const renderWithContext = (component: ReactElement) => {
  const history = createMemoryHistory();
  const wrapper = ({ children }: { children?: ReactNode }) => (
    <Router history={history}>{children}</Router>
  );

  const renderResults = render(component, { wrapper });
  return {
    ...renderResults,
    snapshot: () => {
      expect(renderResults.container).toMatchSnapshot();
    }
  };
};
