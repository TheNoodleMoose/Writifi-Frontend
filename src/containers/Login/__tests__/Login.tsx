import React from "react";
import { renderWithContext } from "../../../testUtils";

import Login from "..";

it("renders correctly", () => {
  renderWithContext(<Login />).snapshot();
});
