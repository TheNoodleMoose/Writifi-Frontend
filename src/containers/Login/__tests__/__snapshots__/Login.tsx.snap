// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <p>
      This is the login page
    </p>
  </div>
</div>
`;
